### DISCUSSION TITLE
Bitwig and Debian Linux users / How to get ctrlr to work

### DISCUSSION THREAD

**Discussion by oliver-ruehl** [RomanKubiak/ctrlr#543] [Q&A]
Hey all,

I just stumbled across this project while looking for a nice way to control the AKAI Miniak (it has a 2 row display which is wheew... well you know).
I'm trying to get this thing running on Bitwig (Debian), but so far I had no luck.

First of all I'm confused about the versions and compatibility of panels (I opened an issue about it already [#issue-1612872693](https://github.com/RomanKubiak/Panels/issues/5#issue-1612872693)

I hope to hear from some people that had more success than me.

Kind regards
Oliver

**Comment by damiensellier** [comment]
Hi Oliver, I replied to your post on the panel repository.
You were talking about using your akai panel in Bitwig Win10 and here on Debian. 
Can you run the Ctrlr software without loading the panel extension? Or the Ctrlr software is crashing from the get go?

[Tobias Schlemmer](https://github.com/keinstein)  has released many updates for Linux https://github.com/keinstein/ctrlr and from what I understand it's working great.

If you could share more about your versions and details about how it crashes that would be helpful

Damien 

PS: if this is the panel you are working with, https://ctrlr.org/akai-miniak-alesis-micron/  I have it working on Mac on both Ctrlr 5.3.198 and 5.6.23  There's a small bug on startup though, nothing problematic.

**Reply by oliver-ruehl** [reply]
I'll dabble around with that version for a bit. I have my Miniak here and some coffee :)
Thanks so much

**Reply by damiensellier** [reply]
https://github.com/RomanKubiak/ctrlr/pull/400 this pull request seems to be working on linux. you should try to compile your own softwares from this fork : https://github.com/keinstein/ctrlr 

Keep us updated about your progress! Hope it will work great.

**Reply by damiensellier** [reply]
Saw your last message but it disappeared. 
Here's the tuto to build Ctrlr on Windows, this will give you some idea : 
https://godlike.com.au/fileadmin/godlike/techtools/ctrlr/guides/Compiling_Ctrlr_for_Windows_10_v2.1.pdf

First step you have to after downloading is unzipping 'boost.zip'. You will find this file in ..\ctrlr-master\Source\Misc\boost\. Just unzip it. 

This is probably why you get errors with luabind and boost.

**Reply by oliver-ruehl** [reply]
ok thanks. Somehow my post disappeared... weird.
I'll try your suggestions later today.

**Comment by oliver-ruehl** [comment]
1. I tried it again with your suggestions about boost.zip and let you know how it goes.
3. I also have JUCE running on Linux and unfortunately the live build tools are still under development.

**Comment by oliver-ruehl** [comment]
I have compiled everything with no errors. I have no clue what to do next.
![Bildschirmfoto_2023-03-12_08-08-18](https://user-images.githubusercontent.com/1587355/224529837-c1e2bd35-642c-4218-a917-c2c0fb279a14.png)

**Reply by oliver-ruehl** [reply]
![image](https://user-images.githubusercontent.com/1587355/224530014-c7ba975e-7759-4be2-b773-440025273b38.png)

The most confusing thing is for me how everything fits together.
I have no compiling experience and it shows :D
I need a mindmap :D

**Comment by oliver-ruehl** [comment]
I get this message when i use the debug VST3

```
/home/elektrik/.vst/Ctrlr-Debug.so

com.bitwig.flt.library.metadata.reader.exception.CouldNotReadMetadataException: could not read metadata: Failed to load VST 2 plug-in /home/elektrik/.vst/Ctrlr-Debug.so: libcurl-gnutls.so.4: cannot open shared object file: No such file or directory
-------------------------------------------------------------------

/home/elektrik/.vst3/Ctrlr-Debug.vst3

com.bitwig.flt.library.metadata.reader.exception.CouldNotReadMetadataException: could not read metadata: Failed to load VST 3 plug-in /home/elektrik/.vst3/Ctrlr-Debug.vst3: libcurl-gnutls.so.4: cannot open shared object file: No such file or directory
-------------------------------------------------------------------

/home/elektrik/neue-vst/Ctrlr-Debug.vst3

com.bitwig.flt.library.metadata.reader.exception.CouldNotReadMetadataException: could not read metadata: Failed to load VST 3 plug-in /home/elektrik/neue-vst/Ctrlr-Debug.vst3: libcurl-gnutls.so.4: cannot open shared object file: No such file or directory
-------------------------------------------------------------------


```

**Reply by damiensellier** [reply]
Sorry mate, I have absolutely no idea how linux works. I can just tell you to try to build CTRLR for "release" and not "debug". 
At this point I would get in touch with Tobias who pulled the Latest Linux updates for CTRLR on this github, I guess he will be able to help you or even share his own builds, you want to make music not programming after all ;)
https://schlemmersoft.de/de/Kontakt
